AzureAD/microsoft-authentication-library-common-for-android
# Fix ignored authority_url when creating the authority audience

## The Problem
When the authority `type` is AAD, `authority_url` value of the `authority` is not being considered when deserializing the msal config object.  This results in the audience created for the authority to use the public cloud. 

## Why is this happening
When deserializing the authority, through the `AuthorityDeserializer` we generate the audience's cloudUrl by invoking `aadAuthority.getAuthorityUri()`. This method is overriden in the `AzureActiveDirectoryAuthority` class to return the cloud URL from the authority. This flows as follows.

1. We deserialize the authority object into an `AzureActiveDirectoryAuthority` object.
2. We try to create the audience  for this object by invoking `getAuthorityUri()` 
3. The `getAuthorityUri()` method builds the authority uri by making use of the audience's cloudUrl (which we are trying to create in the first place)
4. since the audience's cloudUrl is null, it is instantiated to the public cloud and returned as the authority URL.

## Changes made
Revert back to the original implementation before the regression.

## Reproducing the bug

1. Checkout to the `dev` branch on common. 
2. Run MSAL test app.
3. Choose `ARLINGTON` config file. This has its authority_url set to the US Gov cloud.
4. Try to acquire token with device code flow.

###**Expectation**
You should get a message with the device user code as shown below.
![Screenshot_20220606_130755](https://user-images.githubusercontent.com/25629064/172143637-3dbe7f5c-a291-4fb0-b173-c928d3775bc1.png)

###**Actual**
The device code flow fails with an error as shown below.
![Screenshot_20220606_131249](https://user-images.githubusercontent.com/25629064/172143727-6add77f3-42e8-4742-be13-6a8acba04a55.png)


## Testing
1. Checkout to `git/fix-authority-url` branch
2. Run MSAL test app
3. Choose `ARLINGTON` config file. This file has its authority_url set to the US Gov cloud
4. You should be able to acquire token with device code flow successfully. The verification URL should be "https://microsoft.com/deviceloginus"
